libs/ts-rest/express/src/lib/ts-rest-express.ts

```diff
@@ -117,6 +118,45 @@ const recursivelyApplyExpressRouter = (
   }
 };
 
+const validateRequest = (
+  req: Request,
+  res: Response,
+  schema: AppRouteQuery | AppRouteMutation,
+  options: Options
+) => {
+  const paramsResult = checkZodSchema(req.params, schema.pathParams, {
+    passThroughExtraKeys: true,
+  });
+
+  if (!paramsResult.success) {
+    return res.status(400).send(paramsResult.error);
+  }
+
+  const headersResult = checkZodSchema(req.headers, schema.headers, {
+    passThroughExtraKeys: true,
+  });
+
+  if (!headersResult.success) {
+    return res.status(400).send(headersResult.error);
+  }
+
+  const query = options.jsonQuery
+    ? parseJsonQueryObject(req.query as Record<string, string>)
+    : req.query;
+
+  const queryResult = checkZodSchema(query, schema.query);
+
+  if (!queryResult.success) {
+    return res.status(400).send(queryResult.error);
```

Comment:
We'll want to document what these responses will look like when writing the documentation. Most devs will want to know that we'll send a 400 error (and the error string) if the validation does not pass.

Reply:
Yeah, I agree, that makes sense. This is the current behavior already so we'd have to address that in a separate PR.

However, I think the right thing to do would be to do a small refactor to the express implementation where we throw an error, and use a ts-rest-supplied error handler middleware to send the standard zod error that can also be swapped out so the application can send custom error messages.

Reply:
We can technically already do this on the NestJS implementation by using an exception filter and catching `BadRequestException`